library/Vanilla/Embeds/TwitchEmbed.php

```diff
@@ -0,0 +1,198 @@
+    }
+
+    /**
+     * Assigns the link type and retrieves the video id.
+     *
+     * @param $url $string
+     * @return string $videoID
+     */
+    private function getUrlInformation($url): string {
+
+        $domain = parse_url($url, PHP_URL_HOST);
+        $path = parse_url($url, PHP_URL_PATH);
+        $videoID = null;
+
+        if ($domain == "clips.twitch.tv") {
+            $this->urlType = 'clip';
+            preg_match('/\/(?<id>[a-zA_Z0-9_-]+)/i', $path,$clipID);
+            if ($clipID['id']) {
+                $videoID = $clipID['id'];
+            }
+        }
+
+        if ($domain == "www.twitch.tv") {
+            preg_match('/(\/(?<isVideoOrCollection>\w+)\/)?(?<id>[a-zA-Z0-9]+)/i', $path, $linkType);
+            if ($linkType['isVideoOrCollection']) {
+                $this->urlType = ($linkType['isVideoOrCollection'] == 'videos') ? 'video' : 'collection';
+            } else {
+                $this->urlType = 'channel';
+            }
+            $videoID = $linkType['id'];
+        }
+        return $videoID;
+    }
+
+    /**
+     * Gets any query string attached to link.
+     *
+     * @param  string $url
+     * @return array $query
+     */
+    private function getQueryInformation($url): array {
+        $query = [];
+        $queryString = parse_url($url, PHP_URL_QUERY);
+        if ($queryString) {
+            parse_str($queryString, $query);
+        }
+        return $query;
+    }
+
+    /**
+     * Assigns a embed url based on the linktype and query string.
+     *
+     * @param string $videoID
+     * @param array $queryInfo
+     * @return string $embedUrl
+     */
+    private function getEmbedUrl($videoID, $queryInfo = null): string {
+        $embedURL = '';
+        if ($queryInfo) {
+            $t = $this->validateQueryTime($queryInfo['t']);
+            $autoplay = $this->verifyBool($queryInfo['autoplay']);
+            $muted = $this->verifyBool($queryInfo['muted']);
+        }
+
+        if ($this->urlType == 'clip' ) {
+            $embedURL = "https://clips.twitch.tv/embed?clip=".$videoID;
+        }
+
+        if ($this->urlType == 'channel') {
+            $embedURL = "https://player.twitch.tv/?channel=".$videoID;
+        }
+
+        if ($this->urlType == 'collection') {
+            $embedURL = "https://player.twitch.tv/?collection=".$videoID;
+        }
+
+        if ($this->urlType == 'video') {
+            $embedURL = "https://player.twitch.tv/?video=v".$videoID;
+            if ($autoplay) {
+                $embedURL.="&autoplay=".$autoplay;
+            }
+            if ($t) {
+                $embedURL.="&t=".$t;
+            }
+            if ($muted) {
+               $embedURL.="&muted=".$muted;
+            }
+        }
+        return $embedURL;
+    }
+
+    /**
+     * Validates the time parameter of the query string is valid.
+     *
+     * @param string $time
+     * @return string $validTime
+     */
+    private function validateQueryTime($time): string {
```

Comment:
This is certainly not a validation function. After running it I have no way of knowing if it was a valid time string in the first place.

A name like `filterQueryTime` might be more appropriate. Also typo in `$valide`.